props/blaster_BC_buttons.h

```diff
@@ -0,0 +1,788 @@
+    SPEC::SteppedMode::mode_deactivate();
+  }
+
+  void next() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ < max_volume_) {
+      current_volume_ += max_volume_ * 0.10;
+      if (current_volume_ >= max_volume_) {
+        current_volume_ = max_volume_;
+        QuickMaxVolume();
+      } else {
+        mode::getSL<SPEC>()->SayVolumeUp();
+      }
+      dynamic_mixer.set_volume(current_volume_);
+    }
+  }
+
+  void prev() override {
+    int current_volume_ = dynamic_mixer.get_volume();
+    if (current_volume_ > min_volume_) {
+      current_volume_ -= max_volume_ * 0.10;
+      if (current_volume_ <= min_volume_) {
+        current_volume_ = min_volume_;
+        QuickMinVolume();
+      } else {
+        mode::getSL<SPEC>()->SayVolumeDown();
+      }
+      dynamic_mixer.set_volume(current_volume_);
+    }
+  }
+
+  void QuickMaxVolume() {
+    dynamic_mixer.set_volume(max_volume_);
+    PVLOG_NORMAL << "** Maximum Volume\n";
+    mode::getSL<SPEC>()->SayMaximumVolume();
+  }
+
+  void QuickMinVolume() {
+    dynamic_mixer.set_volume(min_volume_);
+    PVLOG_NORMAL << "** Minimum Volume\n";
+    mode::getSL<SPEC>()->SayMinimumVolume();
+  }
+
+  void update() override {  // Overridden to substitute the tick sound
+    float volume = dynamic_mixer.get_volume();
+    percentage_ = round((volume / max_volume_) * 10) * 10;
+    SaberBase::DoEffect(EFFECT_VOLUME_LEVEL, 0);
+  }
+
+  void select() override {
+    PVLOG_NORMAL << "** Saved - Exiting Volume Menu\n";
+    mode::getSL<SPEC>()->SaySave();
+    SPEC::SteppedMode::select();
+  }
+
+  void exit() override {
+    PVLOG_NORMAL << "** Cancelled - Exiting Volume Menu\n";
+    percentage_ = initial_percentage_;
+    dynamic_mixer.set_volume(initial_volume_);
+    mode::getSL<SPEC>()->SayCancel();
+    SPEC::SteppedMode::exit();
+  }
+
+// For Blasters
+  bool mode_Event2(enum BUTTON button, EVENT event, uint32_t modifiers) override {
+    switch (EVENTID(button, event, 0)) {
+
+      case EVENTID(BUTTON_FIRE, EVENT_FIRST_HELD_MEDIUM, 0):  // fire m1
+        QuickMaxVolume();
+        return true;
+      case EVENTID(BUTTON_MODE_SELECT, EVENT_FIRST_HELD_MEDIUM, 0):  // mode m1
+        QuickMinVolume();
+        return true;
+
+      case EVENTID(BUTTON_FIRE, EVENT_FIRST_SAVED_CLICK_SHORT, 0):  // fire s1
+        select();
+        return true;
+      case EVENTID(BUTTON_MODE_SELECT, EVENT_FIRST_SAVED_CLICK_SHORT, 0):  // mode s1
+        exit();
+        return true;
+    }
+    return SPEC::SelectCancelMode::mode_Event2(button, event, modifiers);
+  }
+};
+
+template<class SPEC>
+struct BCBlasterMenuSpec {
+  typedef BCBlasterVolumeMode<SPEC> BCBlasterVolumeMenu;
+  typedef mode::SelectCancelMode SelectCancelMode;
+  typedef mode::SteppedMode<SPEC> SteppedMode;
+  typedef mode::SteppedModeBase<SPEC> SteppedModeBase;
+  typedef mode::MenuBase<SPEC> MenuBase;
+  typedef SoundLibraryV2 SoundLibrary;
+};
+
+  void Setup() {
```

Comment:
Why is this global, and indented?

Reply:
Do you mean it's global, so fix the indent, or is it wrong to be global?

Reply:
I think it should be inside the prop. (and I think you want an "override" on it)

Reply:
Done, thanks. Not too aware of the startup routine yet.